src/modules/services/prometheus.nix

```
@@ -0,0 +1,185 @@
+      };
+
+      retentionTime = lib.mkOption {
+        type = lib.types.str;
+        default = "15d";
+        description = "How long to retain data";
+      };
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 9090;
+      description = "Port for Prometheus web interface";
+    };
+
+    globalConfig = lib.mkOption {
+      type = lib.types.attrs;
+      default = {
+        scrape_interval = "1m";
+        scrape_timeout = "10s";
+        evaluation_interval = "1m";
+      };
+      description = "Global Prometheus configuration";
+    };
+
+    scrapeConfigs = lib.mkOption {
+      type = lib.types.listOf lib.types.attrs;
+      default = [ ];
+      description = "List of scrape configurations";
+    };
+
+    ruleFiles = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      description = "List of rule files to load";
+    };
+
+    alerting = lib.mkOption {
+      type = lib.types.nullOr lib.types.attrs;
+      default = null;
+      description = "Alerting configuration";
+    };
+
+    remoteWrite = lib.mkOption {
+      type = lib.types.listOf lib.types.attrs;
+      default = [ ];
+      description = "Remote write configurations";
+    };
+
+    remoteRead = lib.mkOption {
+      type = lib.types.listOf lib.types.attrs;
+      default = [ ];
+      description = "Remote read configurations";
+    };
+
+    experimentalFeatures = {
+      enableExemplars = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = "Enable exemplar storage";
+      };
+
+      enableTracing = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = "Enable tracing";
+      };
+
+      enableOTLP = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = "Enable OTLP receiver";
+      };
+    };
+
+    advanced = {
+      storage = lib.mkOption {
+        type = lib.types.attrs;
+        default = { };
+        description = "Storage configuration";
+      };
+
+      tsdb = lib.mkOption {
+        type = lib.types.attrs;
+        default = { };
+        description = "TSDB configuration";
+      };
+    };
+
+    extraArgs = lib.mkOption {
+      type = lib.types.str;
+      default = "";
+      description = "Additional arguments to pass to Prometheus";
+    };
+
+    configFile = lib.mkOption {
+      type = lib.types.path;
+      default = configFile;
+      internal = true;
+      description = "The generated Prometheus configuration file";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    packages = [
+      cfg.package
+    ];
+
+    processes.prometheus = {
+      exec = "${cfg.package}/bin/prometheus ${prometheusArgs}";
+
+      process-compose = {
+        readiness_probe = {
+          http_get = {
+            host = "127.0.0.1";
+            port = cfg.port;
+            path = "/-/ready";
+          };
+          initial_delay_seconds = 5;
```

```suggestion
          initial_delay_seconds = 2;
```
That seems excessively long for a Go server startup.